Ability to directly move an item to another directory

oleitao opened the pull request:
<!-- 
🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
I ACKNOWLEDGE THE FOLLOWING BEFORE PROCEEDING:
1. PR may be deleted if it is not following the template
2. Only post one request per one feature request
3. Try not to make duplicates. Do a quick search before posting
4. Add a clarified title
-->

**Details of Changes**
Add details of changes here.
- Added the ability to directly move an item to another directory

Closes #946

**Validation**
How did you test these changes?
- [x] Select an folder and move into another folder


![test1](https://user-images.githubusercontent.com/72361786/115130618-07ae6c80-9fe9-11eb-81dc-99edccb37106.gif)

yaichenbaum commented:
@oleitao It looks like the screen recording got messed up, can you try recording it again?

oleitao commented:
> @oleitao It looks like the screen recording got messed up, can you try recording it again?

@yaichenbaum ofcourse i can

![test1](https://user-images.githubusercontent.com/72361786/115155291-23605400-a077-11eb-9c77-6b894eb0237b.gif)

yaichenbaum commented:
@oleitao Can you link to an issue?

oleitao commented:
> @oleitao Can you link to an issue?

@yaichenbaum i don't have permission to link my PR

yaichenbaum commented:
@oleitao You can add it to the description https://github.com/blog/957-introducing-issue-mentions.

oleitao commented:
> @oleitao You can add it to the description https://github.com/blog/957-introducing-issue-mentions.

I fixed issue in [#946](https://github.com/files-community/Files/issues/946)

yaichenbaum commented:
@oleitao You can add `Closes #946` to the PR description to link to the issue.

oleitao commented:
> @oleitao You can add `Closes #946` to the PR description to link to the issue.

How do i do that?

yaichenbaum commented:
> > @oleitao You can add `Closes #946` to the PR description to link to the issue.
> 
> How do i do that?

I just added it for you.
![image](https://user-images.githubusercontent.com/39923744/116300453-eaa63600-a76c-11eb-8e15-105b66953031.png)

oleitao commented:
> > > @oleitao You can add `Closes #946` to the PR description to link to the issue.
> > 
> > 
> > How do i do that?
> 
> I just added it for you.
> ![image](https://user-images.githubusercontent.com/39923744/116300453-eaa63600-a76c-11eb-8e15-105b66953031.png)

**Details of Changes**

- Added on right-click ContextMenu an option command 'Move...' to move from an folder to another selected by user.

Closes [#946](https://github.com/files-community/Files/issues/946)

**Validataion**

- Select an folder to move into another destination
- Select 'Move...' option
- Select on DialogBox destination folder to move
- Navigate into destination folder and check if selected folder was correctly moved (and it's content)
- Navigate into origin folder and check if selected folder doesn't exist.

![Capture1](https://user-images.githubusercontent.com/72361786/116316965-9de15000-a7aa-11eb-9327-99b5ff6bda58.PNG)

oleitao commented:
@yaichenbaum i would like to resolve conflicts in order to finish this PR. However, i'm not allowed to do that
![Capture3](https://user-images.githubusercontent.com/72361786/117665077-61462900-b19a-11eb-8ad0-6427695b1373.PNG)

yaichenbaum commented:
@oleitao That's because there are too many conflicts to resolve through GitHub, you'll need to resolve them locally.

tsvietOK commented on Files/Interacts/BaseLayoutCommandImplementationModel.cs:
```diff
@@ -564,9 +564,14 @@ public void SearchUnindexedItems(RoutedEventArgs e)
-        public async void CreateFolderWithSelection(RoutedEventArgs e)
+        public virtual async void MoveItem(RoutedEventArgs e)
         {
-            UIFilesystemHelpers.CreateFolderWithSelectionAsync(associatedInstance);
```
Why did you remove `CreateFolderWithSelection` method?

tsvietOK commented:
Looks like this function is not working with shortcut files, so i guess, `Move` menu item shouldn't be visible for shortcuts.

tsvietOK commented:
When pressing Move button app crashes.
By the way, to hide Move menu item for shortcuts, just use the next:
```cs
{
    Text = "BaseLayoutItemContextFlyoutMove/Text".GetLocalized(),
    Glyph = "\uE8B4",
    Command = commandsViewModel.MoveItemCommand,
    ShowItem = selectedItems.All(x => !x.IsShortcutItem)
},

```
Please, test app properly before requesting review.

yaichenbaum commented:
Closing this PR due to inactivity.
